corehq/apps/commtrack/consumption.py

```diff
@@ -17,11 +20,44 @@ def recalculate_domain_consumption(domain):
     products = Product.by_domain(domain)
     for supply_point_id in found_doc_ids:
         for product in products:
-            filtered_transactions = StockTransaction.get_ordered_transactions_for_stock(
-                supply_point_id, const.SECTION_TYPE_STOCK, product._id
-            )
-            if filtered_transactions:
-                update_stock_state_for_transaction(filtered_transactions[0])
+            try:
+                latest_transaction = StockTransaction.get_ordered_transactions_for_stock(
+                    supply_point_id, const.SECTION_TYPE_STOCK, product._id
+                )[0]
+            except IndexError:
+                pass
+            else:
+                state = get_stock_state_for_transaction(latest_transaction)
+                daily_consumption = get_consumption_for_ledger(state)
+                state.daily_consumption = daily_consumption
+                with drop_connected_signals(post_save):
```

Comment:
One thing that just occurred to me is what happens to other save's while the signal is dropped. I assume that they'll be in a separate 'thread' and hence unaffected but could be worth testing before this gets merged.